### PR TITLE
sync umgr and pmgr

### PR DESCRIPTION
**Pull request by andre-merzky**
I merged #2013 before tests passed - mea culpa Matteo!  This fixes the problem which was caused by umgr and pmgr getting out of sync, by ensuring their construction and initialization follows the same sequence.  A diff between `unit_manager.py` and `pilot_manager.py` will motivate the changes in this PR.

### CODE REVIEW

**Comment by andre-merzky** [comment]
Hey @mturilli : I fixed some tests (spaces *urgh*), but the tests still  fail because of #2002.  The umgr (point of this PR) seems clean now.

File: src/radical/pilot/pilot_manager.py [hunk 1]
```diff
@@ -150,14 +152,16 @@ def __init__(self, session, cfg='default'):
...
-    def initialize_common(self):
+    def initialize(self):
 
         # the manager must not carry bridge and component handles across forks
         ru.atfork(self._atfork_prepare, self._atfork_parent, self._atfork_child)
 
 
     # --------------------------------------------------------------------------
     #
+    # EnTK forks, make sure we don't carre traces of children across the fork
```
Comment (mturilli) [inline src/radical/pilot/pilot_manager.py]: Just a typo: carre -> carry

File: src/radical/pilot/pilot_manager.py [hunk 2]
```diff
@@ -198,7 +193,7 @@ def close(self, terminate=True):
         self._terminate.set()
         self._rep.info('<<close pilot manager')
 
-        # we don't want any callback invokations during shutdown
+        # disable callbacks during shutdown
         # FIXME: really?
```
Comment (mturilli) [inline src/radical/pilot/pilot_manager.py]: Found this funny. Why really?

File: src/radical/pilot/unit_manager.py [hunk 3]
```diff
@@ -152,11 +159,38 @@ def __init__(self, session, scheduler=None):
...
+    #
+    def initialize(self):
+
+        # the manager must not carry bridge and component handles across forks
+        ru.atfork(self._atfork_prepare, self._atfork_parent, self._atfork_child)
+
+
+    # --------------------------------------------------------------------------
+    #
+    # EnTK forks, make sure we don't carre traces of children across the fork
```
Comment (mturilli) [inline src/radical/pilot/unit_manager.py]: carre -> carry

File: src/radical/pilot/pilot_manager.py (hunk 2 above)
Comment (andre-merzky) [reply src/radical/pilot/pilot_manager.py]: In one way, I don't want to get notified ' you are terminating' when I initiate a termination - that is redundant.  And we need to somewhat carefully check that we don't run circles during termination (callback encountering termination and thus trying to terminate...).  Other than that I am not sure - we should eventually revisit that.

Thanks for the typos, will fix.

**Pull request merged**